skrifa/src/meta/provider.rs

```diff
@@ -0,0 +1,25 @@
+use super::metrics::{GlyphMetrics, Metrics};
+use crate::NormalizedCoord;
+
+/// Interface for types that can provide font metadata.
+pub trait MetadataProvider<'a>: raw::TableProvider<'a> + Sized {
```

Comment:
This is currently just metrics, not any broader metadata; I wonder if it should be narrowed? If not perhaps we could explain what we anticipate the scope growing to be.

This is just personal preference but ... I kind of like the narrowness and would personally be a fan of it becoming more specific, e.g. `MetricReader` (MetricFactory in Java terms, unless we can somehow get to MetricFactoryFactory which would earn +20 pts)

Reply:
The code I have staged for full metadata has many more methods here for attributes, charmaps, strings, axes and instances (so far). Rust requires you to import traits to make use of the methods so I think breaking them up adds friction for the user for no appreciable gain. I’d prefer to keep it as is for now and revisit after the full set lands.